pkg/scheduler/scheduler.go

```diff
@@ -780,11 +781,29 @@ type podConditionUpdaterImpl struct {
 
 func (p *podConditionUpdaterImpl) update(pod *v1.Pod, condition *v1.PodCondition) error {
 	klog.V(3).Infof("Updating pod condition for %s/%s to (%s==%s, Reason=%s)", pod.Namespace, pod.Name, condition.Type, condition.Status, condition.Reason)
-	if podutil.UpdatePodCondition(&pod.Status, condition) {
-		_, err := p.Client.CoreV1().Pods(pod.Namespace).UpdateStatus(context.TODO(), pod, metav1.UpdateOptions{})
-		return err
-	}
-	return nil
+	// First try to update using the pod that was passed in. If conflict occurs, get latest pod and try again.
+	updatePodConditionBackoff := wait.Backoff{
+		Steps:    5,
+		Duration: 100 * time.Millisecond,
+		Jitter:   1.0,
+	}
+	firstTry := true
+	return clientretry.RetryOnConflict(updatePodConditionBackoff, func() error {
+		if firstTry {
+			firstTry = false
+		} else {
+			var err error
+			pod, err = p.Client.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
```

Comment:
```suggestion
			pod, err = p.getUpdatedPod(pod)
```

Reply:
I was going to do this, but there is no getUpdatedPod on podConditionUpdaterImpl, only on podPreemptorImpl.  Should I add it?

Reply:
I went ahead and added it.

Reply:
Ah, I misread - thought they were of the same type. Then nvm, it's not worth to make it a function only for one usage. Feel free to revert to the previous version.